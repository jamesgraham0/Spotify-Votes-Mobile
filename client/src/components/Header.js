import { StyleSheet, Text, View, TouchableOpacity} from 'react-native';
import React, { useState, useEffect } from 'react';
import Constants from '../utils/constants';
import Ionicons from '@expo/vector-icons/Ionicons';

export default Header = ({ headerText, onBackPress }) => {
    return (
        <View style={Constants.HEADER_STYLES}>
            <TouchableOpacity onPress={onBackPress} style={styles.returnButton}>
                <Ionicons name="chevron-back-circle-outline" size={32} color="grey" />
            </TouchableOpacity>
            <Text style={styles.headerText}>{headerText}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    headerText: {
        position: 'absolute',
        top: '50%',
        color: '#fff',
        fontSize: 24,
        maxWidth: '70%',
    },
    returnButton: {
        position: 'absolute',
        top: '50%',
        left: 20,
        width: 50,
        height: 50,
    },
});