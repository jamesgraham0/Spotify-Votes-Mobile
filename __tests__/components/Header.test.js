import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../client/src/components/Header';

describe('Header', () => {
    it('should render the Header component', () => {
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});