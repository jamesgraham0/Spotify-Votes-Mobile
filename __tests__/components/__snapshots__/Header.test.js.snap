// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the Header component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(10, 10, 10, 0.5)",
      "flexDirection": "row",
      "height": "13%",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 50,
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": "50%",
        "width": 50,
      }
    }
  >
    <Text />
  </View>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 24,
        "maxWidth": "70%",
        "position": "absolute",
        "top": "50%",
      }
    }
  />
</View>
`;
